src/connectors/hierarchical-menu/__tests__/connectHierarchicalMenu-test.js

```diff
@@ -809,120 +836,491 @@ See documentation: https://www.algolia.com/doc/api-reference/widgets/hierarchica
+        {
+          name: 'categoriesLvl0',
+          attributes: ['categoriesLvl0', 'categoriesLvl1'],
+          separator: ' > ',
+          rootPath: null,
+          showParentLevel: true,
+        },
+      ]);
+
+      expect(actual.hierarchicalFacetsRefinements).toEqual({
+        categoriesLvl0: ['TopLevel > SubLevel'],
+      });
+    });
+
+    test('returns the `SearchParameters` with a custom `separator`', () => {
+      // Uses the function getInitializedWidget once we've removed `getConfiguration`
+      const render = () => {};
+      const makeWidget = connectHierarchicalMenu(render);
+      const helper = jsHelper({}, '');
+      const widget = makeWidget({
+        attributes: ['categoriesLvl0', 'categoriesLvl1'],
+        separator: ' / ',
+      });
+
+      const actual = widget.getWidgetSearchParameters(helper.state, {
+        uiState: {},
+      });
+
+      expect(actual.hierarchicalFacets[0].separator).toBe(' / ');
+    });
+
+    test('returns the `SearchParameters` with a custom `rootPath`', () => {
+      // Uses the function getInitializedWidget once we've removed `getConfiguration`
+      const render = () => {};
+      const makeWidget = connectHierarchicalMenu(render);
+      const helper = jsHelper({}, '');
+      const widget = makeWidget({
+        attributes: ['categoriesLvl0', 'categoriesLvl1'],
+        rootPath: 'TopLevel > SubLevel',
+      });
+
+      const actual = widget.getWidgetSearchParameters(helper.state, {
+        uiState: {},
+      });
+
+      expect(actual.hierarchicalFacets[0].rootPath).toBe('TopLevel > SubLevel');
+    });
+
+    test('returns the `SearchParameters` with a custom `showParentLevel`', () => {
+      // Uses the function getInitializedWidget once we've removed `getConfiguration`
+      const render = () => {};
+      const makeWidget = connectHierarchicalMenu(render);
+      const helper = jsHelper({}, '');
+      const widget = makeWidget({
+        attributes: ['categoriesLvl0', 'categoriesLvl1'],
+        showParentLevel: true,
+      });
+
+      const actual = widget.getWidgetSearchParameters(helper.state, {
+        uiState: {},
+      });
+
+      expect(actual.hierarchicalFacets[0].showParentLevel).toBe(true);
+    });
+
+    it('warns with the same `hierarchicalFacets` already defined with different `attributes`', () => {
+      const render = () => {};
+      const makeWidget = connectHierarchicalMenu(render);
+      const helper = jsHelper({}, '', {
+        hierarchicalFacets: [
+          {
+            name: 'category',
+            attributes: ['category', 'sub_category', 'sub_sub_category'],
+            separator: ' > ',
+            rootPath: null,
+          },
+        ],
+      });
+
+      const widget = makeWidget({
+        attributes: ['category', 'sub_category'],
+      });
+
+      expect(() =>
+        widget.getWidgetSearchParameters(helper.state, {
+          uiState: {},
+        })
+      ).toWarnDev();
```

Comment:
Should we track the warning message?

Reply:
We have only one possible message at the moment, not sure it is worth it.